staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -360,6 +360,9 @@ func finishNothing(context.Context, bool) {}
 
 // Create inserts a new item according to the unique key from the object.
 func (e *Store) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	// Make a copy so we never mutate the passed-in object (e.g. in tests)
+	obj = obj.DeepCopyObject()
```

Comment:
Would be a regression in hot path serving performance - Create in apiserver should be called with net new objects anyway.  Is this ever a problem from outside of tests (someone calling storage create path in another storage call)?

Reply:
I don't think it's a REAL problem, just in tests.  Ergo my email to sig-api-machinery.

I don't think it's realistic to ask every test to pass a DeepCopy()

Reply:
Why?  If you call create you need to understand what Create does.  You're giving it an object and get another object in return.  We could make it explicit that it's unsafe like Convert (UnsafeConvert mutates, Convert copies, everyone uses UnsafeConvert in hot path), which is our extant pattern for this concern.

Note that serialization mutates right now too...

Reply:
The more we disperse the responsibility, the more likely someone is to flub it.

IF we set a flag in `Store` that indicated "testing mode" then we could make this conditional.  That still depends on tests setting the flag, but that will get cargo-culted over time.  I dislike having test-only code in prod, but it's better than nothing?

Reply:
FWIW we do have the `notest` build tag set on production binary builds so we could have this gated at compile time by that (ref:https://github.com/kubernetes/kubernetes/pull/92491)

I'm not sure that's the right answer, however.